include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_scan_by_segment.h

```diff
@@ -1,3 +1,5 @@
+// -*- C++ -*-
+//===-- parallel_backend_sycl_scan_by_segment.h ---------------------------===//
```

Comment:
Last couple of years I am sharing a rule "no filename duplication in a file header".
Because, 
1) This information is redundant. A developer can see a filename in any case (in a tab or in a file tree).
2) Writing a filename might lead to "desynchronization" in case of filer naming - a developer rather forget to update a filename in this header... Even if he doesn't forget - it is double effort in any case.
3) A did not get any objects yet regarding that rule.

Reply:
My opinion - it just doesn't make sense to discuss about the things like this.
At this moment we have ~54 entries in our includes like that.
If author prefer to write it - let's go, no problems.

Reply:
I do not have an issue if we want this to be the convention going forward, so I made this change. I also made some other small tweaks to the header comment to be consistent with other files.